projects/igniteui-angular/src/lib/directives/toggle/toggle.directive.ts

```diff
@@ -229,28 +227,29 @@ export class IgxToggleDirective implements IToggleView, OnInit, OnDestroy {
         if (!this.collapsed && this._overlayId) {
             this.overlayService.hide(this._overlayId);
         }
-        if (this._overlayClosedSub) {
-            if (!this._overlayClosedSub.closed) {
-                this._overlayClosedSub.unsubscribe();
-            }
-        }
-        if (this._overlayClosingSub) {
-            if (!this._overlayClosingSub.closed) {
-                this._overlayClosingSub.unsubscribe();
-            }
-        }
+        this.unsubscribe();
     }
 
     private overlayClosed = () => {
         this._collapsed = true;
         this.cdr.detectChanges();
+        this._overlayId = undefined;
```

Comment:
If you need to clear the overlayId just use -> delete this._overlayId

Reply:
fixed